samples/core/src/test/java/io/oasp/gastronomy/restaurant/staffmanagement/logic/impl/StaffmanagementImplTest.java

```diff
@@ -0,0 +1,118 @@
+package io.oasp.gastronomy.restaurant.staffmanagement.logic.impl;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import io.oasp.gastronomy.restaurant.staffmanagement.dataaccess.api.StaffMemberEntity;
+import io.oasp.gastronomy.restaurant.staffmanagement.dataaccess.api.dao.StaffMemberDao;
+import io.oasp.gastronomy.restaurant.staffmanagement.logic.api.to.StaffMemberEto;
+import io.oasp.module.beanmapping.common.api.BeanMapper;
+import io.oasp.module.test.common.base.ModuleTest;
+
+/**
+ * This class provides a basic implementation of a {@code ModuleTest} which employs the Mockito framework to provide
+ * fake objects to the SUT. For a limited number of test cases per class it can be convenient to use the annotation
+ * {@code @InjectMocks} to inject all needed mocks in the class under test.
+ *
+ * @author sroeger
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class StaffmanagementImplTest extends ModuleTest {
+
+  @Mock
+  private BeanMapper beanMapper;
+
+  @Mock
+  private StaffMemberEntity staffMemberEntity;
+
+  @Mock
+  private StaffMemberEto staffMemberEto;
+
+  @Mock
+  private StaffMemberDao staffMemberDao;
+
+  @InjectMocks
+  StaffmanagementImpl staffmanagementImpl = new StaffmanagementImpl();
+
+  /**
+   * This test method tests the {@code findStaffMember} method of {@link StaffmanagementImpl} by mocking needed
+   * dependencies.
+   */
+  @Test
+  public void testfindStaffMember() {
+
+    // setup: availability of mocked dependencies is verified
+    assertThat(this.staffMemberEntity).isNotNull();
+    assertThat(this.staffMemberEto).isNotNull();
+    assertThat(this.staffMemberDao).isNotNull();
+    assertThat(this.beanMapper).isNotNull();
+
+    // given
+    long id = 1L;
+    Class<StaffMemberEto> targetClass = StaffMemberEto.class;
+
+    Mockito.when(this.staffMemberDao.find(id)).thenReturn(this.staffMemberEntity);
+    Mockito.when(this.beanMapper.map(this.staffMemberEntity, targetClass)).thenReturn(this.staffMemberEto);
+
+    // when
+
+    StaffMemberEto resultEto = this.staffmanagementImpl.findStaffMember(id);
+
+    // then
+    Mockito.verify(this.staffMemberDao).find(id);
+    Mockito.verify(this.beanMapper).map(this.staffMemberEntity, targetClass);
```

Comment:
This verifications do not contain any added value, do they? The mocked functions has to be called anyhow to reach the asserted result beyond.

Verification of method calls have rather rare use cases. E.g. you want to assure, that statistics are called to be written without actually testing them. The call will not have any impact on the result to be asserted anyhow. But you want to test, that statistics are actually called to be written.